src/tests/golden/quickfix_test.go

```diff
@@ -0,0 +1,126 @@
+package golden
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"github.com/VKCOM/noverify/src/linttest"
+)
+
+func TestQuickFix(t *testing.T) {
+	s := NewQuickFixTest(t, `testdata/quickfix`)
+	s.runQuickFixTest()
+}
+
+const (
+	expectedExtension = ".fix.expected"
+	fixExtension      = ".fix"
+)
+
+type QuickFixTest struct {
+	t      *testing.T
+	folder string
+}
+
+func NewQuickFixTest(t *testing.T, folder string) QuickFixTest {
+	return QuickFixTest{
+		t:      t,
+		folder: folder,
+	}
+}
+
+func (t *QuickFixTest) runQuickFixTest() {
+	files, err := linttest.FindPHPFiles(t.folder)
+	if err != nil {
+		t.t.Errorf("Error while searching for files in the %s folder: %s", t.folder, err)
+	}
+
+	for _, file := range files {
+		t.t.Run(strings.TrimSuffix(filepath.Base(file), ".php"), func(t *testing.T) {
+			testFileName := file
+			expectedFileName := file + expectedExtension
+			fixedFileName := file + fixExtension
+
+			testFile, err := os.Open(testFileName)
```

Comment:
If you're trying to read everything from a file, use `ioutil.ReadFile` instead of Open+ReadAll.
Unless you want a stream-like file handling (but you still do ReadAll right after the close, so it's not the case).